app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -110,6 +113,18 @@ open class FenixApplication : Application() {
         registerActivityLifecycleCallbacks(visibilityLifecycleCallback)
 
         components.core.sessionManager.register(NotificationSessionObserver(this))
+
+        if (!DateUtils.isToday(this.settings.lastPlacesStorageMaintenance)) {
```

Comment:
I'd write this as "if (last maintenance happened more than 24hrs ago) {}". I _think_ `isToday` does that..?